seatunnel-common/src/main/java/org/apache/seatunnel/common/exception/CommonErrorCode.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.common.exception;
+
+public enum CommonErrorCode implements SeaTunnelErrorCode {
+    ENUM_ILLEGAL("Common-01", "This enum is not supported by SeaTunnel"),
+    FILE_OPERATION_ERROR("Common-02", "File operation error"),
+    JSON_OPERATION_ERROR("Common-03", "Json covert/parse operation error"),
+    REFLECT_CLASS_OPERATION_ERROR("Common-04", "Reflect class operation error"),
+    SERIALIZE_OPERATION_ERROR("Common-05", "Serialize class operation error"),
+    UNSUPPORTED_OPERATION("Common-06", "Unsupported operation error");
```

Comment:
Such a design may be more suitable for client users, e.g. http-server <-> app-client

Reply:
> Such a design may be more suitable for client users, e.g. http-server <-> app-client

+1

Reply:
> Such a design may be more suitable for client users, e.g. http-server <-> app-client

For some of our users it might base on SeaTunnel to build some config files and pull a SeaTunnel task. Most application scenarios are based on the client side to collect logs, clear code hints help peripheral integration framework to better collect information. In the actual implementation experience, clear code hints help better user experience, this is also our previous experience in the implementation of DataX to the production environment, whether it is web development, or data integration framework, the ultimate goal is to give users a better experience.

Reply:
> > Such a design may be more suitable for client users, e.g. http-server <-> app-client
> 
> For some of our users it might base on SeaTunnel to build some config files and pull a SeaTunnel task. Most application scenarios are based on the client side to collect logs, clear code hints help peripheral integration framework to better collect information. In the actual implementation experience, clear code hints help better user experience, this is also our previous experience in the implementation of DataX to the production environment, whether it is web development, or data integration framework, the ultimate goal is to give users a better experience.

This is more appropriate in the seatunnel web,because there are more error scenarios in the application system.This is my personal opinion.

Reply:
Sorry, I didn't aggre with you. SeaTunnel without web it is also a independent project. User can create config file by themselves, if the config file created by users  has some errors such as config option not correctly or this task matched some exceptions how to tip user? Throw the exceptions we defined? What should user do if them not understand these exceptions? I think it's not a good idea.